code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm

```diff
@@ -373,14 +365,13 @@
 		emagged = TRUE
 		locked = SMART_FRIDGE_LOCK_SHORTED
 
-/obj/machinery/smartfridge/secure/Topic(href, href_list)
+/obj/machinery/smartfridge/secure/tgui_act(action, params)
 	if(stat & (BROKEN|NOPOWER))
```

Comment:
Here it *is* possible to use the buttons here since the `..()` check isnt at the start. Someone *could* be able to attempt to eject an item from range, but the parent call at the bottom will stop it. So they can be __denied__ at range, but not granted.

Personally I would take the check below and just bolt it into the regular smartfridge `tgui_act`, while checking for `is_secure` in this proc, but this *should* be fine. Up to you really.

Reply:
done